daffodil-core/src/main/scala/org/apache/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -654,6 +655,36 @@ trait SequenceRuntimeValuedPropertiesMixin
     }
   }
 
+  def checkDelimiterEscapeConflict(childTerm: Term): Unit = {
+    if (childTerm.optionEscapeScheme.isDefined &&
+        (childTerm.optionEscapeScheme.get.escapeKind == EscapeKind.EscapeCharacter) &&
+        (hasTerminator || hasSeparator)) {
+      val ecEv = childTerm.optionEscapeScheme.get.escapeCharacterEv
+      if (ecEv.isConstant) {
```

Comment:
This only checks for constant escape characters and constant terminators/separators. And it only checks those defined on the same element. Do we also need to check all in-scope terminating markup? Which really can only easilyl be checked at runtime? Not sure how much overhead that might have though?

Reply:
I did some digging into adding runtime checks, which on the unparsing side was easy.  Parsing, however, did not seem straightforward.  It wouldn't be too bad to add checks agains initiators and terminators, but there is kind of a gap with how the escape scheme is part of the individual elements but the separator is part of the parent sequence.  I might be missing something obvious, but I didn't see a straight forward solution to this that wouldn't involve significant changes.

Reply:
Maybe we should try to match IBM DFDL behavior? Could we write some tests and see how they handle conflicts with runtime escape schemes/delimiters? Can also determine how they handle EEC as well.